File: AndroidStudio/Calculadora/app/src/main/java/com/example/c192/MainActivity.java
package com.example.c192;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Recibir los elementos del layout
        EditText txt1 = findViewById(R.id.num1);
        EditText txt2 = findViewById(R.id.num2);
        TextView txt3 = findViewById(R.id.resul);

        Button btnSumar = findViewById(R.id.sumar);
        Button btnRestar = findViewById(R.id.restar);
        Button btnMultiplicar = findViewById(R.id.multiplicar);
        Button btnDividir = findViewById(R.id.dividir);

        // Función común para manejar operaciones
        View.OnClickListener operar = new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String num1 = txt1.getText().toString().trim();
                String num2 = txt2.getText().toString().trim();

                if (num1.isEmpty() || num2.isEmpty()) {
                    Toast.makeText(getApplicationContext(), "INGRESE TODOS LOS VALORES", Toast.LENGTH_LONG).show();
                    return;
                }

                try {
                    int numero1 = Integer.parseInt(num1);
                    int numero2 = Integer.parseInt(num2);
                    int resultado = 0;

                    switch (view.getId()) {
                        case R.id.sumar:
                            resultado = numero1 + numero2;
                            break;
                        case R.id.restar:
                            resultado = numero1 - numero2;
                            break;
                        case R.id.multiplicar:
                            resultado = numero1 * numero2;
                            break;
                        case R.id.dividir:
                            if (numero2 != 0) {
                                resultado = numero1 / numero2;
                            } else {
                                Toast.makeText(getApplicationContext(), "No se puede dividir por cero", Toast.LENGTH_LONG).show();
                                return;
                            }
                            break;
                    }

                    txt3.setText("El resultado es: " + resultado);
                    // Limpiar los campos después de la operación
                    txt1.setText("");
                    txt2.setText("");
                } catch (NumberFormatException e) {
                    Toast.makeText(getApplicationContext(), "Ingrese valores numéricos válidos", Toast.LENGTH_LONG).show();
                    Log.e("Error", "Error al convertir texto a número", e);
                }
            }
        };

        // Asignar el mismo evento a todos los botones
        btnSumar.setOnClickListener(operar);
        btnRestar.setOnClickListener(operar);
        btnMultiplicar.setOnClickListener(operar);
        btnDividir.setOnClickListener(operar);
    }
}

File: AndroidStudio/Individual6M5/app/src/main/java/com/example/individual6m5/MainActivity.kt
package com.example.individual6m5

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.math.RoundingMode
import java.text.NumberFormat
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val resultTextView = findViewById<TextView>(R.id.resultTextView)
        val sumButton = findViewById<Button>(R.id.btnSum)
        val roundButton = findViewById<Button>(R.id.btnRound)
        val formatButton = findViewById<Button>(R.id.btnFormat)

        // Botón para la suma
        sumButton.setOnClickListener {
            val result = suma(10.5, 5.0f)
            resultTextView.text = "Resultado de la suma: $result"
        }

        // Botón para redondear
        roundButton.setOnClickListener {
            val result = redondeo(15.73)
            resultTextView.text = "Resultado redondeado: $result"
        }

        // Botón para mostrar en formato de moneda
        formatButton.setOnClickListener {
            val result = muestraResultado(1234.56)
            resultTextView.text = "Formato de moneda: $result"
        }
    }

    // Función suma
    private fun suma(a: Double, b: Float): Double {
        return a + b
    }

    // Función redondeo
    private fun redondeo(num: Double): Double {
        return num.toBigDecimal().setScale(0, RoundingMode.UP).toDouble()
    }

    // Función resultado en formato de moneda
    private fun muestraResultado(num: Double): String {
        val format = NumberFormat.getCurrencyInstance(Locale.getDefault())
        return format.format(num)
    }
}